src/Illuminate/Database/Console/AbstractDatabaseCommand.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Illuminate\Database\Console;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Illuminate\Console\Command;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Support\Arr;
+use Symfony\Component\Process\Exception\ProcessSignaledException;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Symfony\Component\Process\Process;
+
+abstract class AbstractDatabaseCommand extends Command
+{
+    /**
+     * Get a human-readable platform name.
+     *
+     * @param  \Doctrine\DBAL\Platforms\AbstractPlatform  $platform
+     * @param  string  $database
+     * @return string
+     */
+    protected function getPlatformName(AbstractPlatform $platform, $database)
+    {
+        return match (class_basename($platform)) {
+            'MySQLPlatform' => 'MySQL <= 5',
+            'MySQL57Platform' => 'MySQL 5.7',
+            'MySQL80Platform' => 'MySQL 8',
+            'PostgreSQL100Platform', 'PostgreSQLPlatform' => 'Postgres',
+            'SqlitePlatform' => 'SQLite',
+            'SQLServerPlatform' => 'SQL Server',
+            'SQLServer2012Platform' => 'SQL Server 2012',
+            default => $database,
+        };
+    }
+
+    /**
+     * Get the size of a table in bytes.
+     *
+     * @param  \Illuminate\Database\ConnectionInterface  $connection
+     * @param  string  $table
+     * @return null
+     */
+    protected function getTableSize(ConnectionInterface $connection, string $table)
+    {
+        return match (class_basename($connection)) {
```

Comment:
I would do `match(true) { $connection instanceof ConnectionClass => ...` to avoid issues with custom connections.